lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -14,7 +16,7 @@ def self.raise_ems_event(event)
     aevent.merge!('Host::host'            => event.src_host.id,            :host_id      => event.src_host.id)            unless event.src_host.nil?
     aevent.merge!('Host::dest_host'       => event.dest_host.id,           :dest_host_id => event.dest_host.id)           unless event.dest_host.nil?
 
-    MiqAeEngine.deliver(event.class.name, event.id, aevent, 'Event')
+    MiqAeEngine.deliver(event.class.name, event.id, aevent, 'Event', user.id)
```

Comment:
@lfu @gmcculloug 
This code needs to be refactored to use the MiqAeEngine.resolve_automation_object which is a synchronous call. The way deliver is being used it uses legacy positional arguments instead of the newer hash format which is being used elsewhere in th workflow when we make a direct synchronous call into Automate.